### PR TITLE
Close mobile menu on link click

### PR DESCRIPTION
**Pull request by nmauersberg**
This pr adds the functionality to close the mobile menu overlay, when an action is clicked.

Also mobile and desktop menu components are now in separate folders to make it a bit more clear, which are which.

A small enhancement of the dynamic actions sneaked into this PR, it was not intentional, but the commit must have went in. Maybe you could have a look on that too @Valioesi ? I left a comment where it was in the code.

### CODE REVIEW

**Review by schaerta — CHANGES_REQUESTED** [review]
It looks great, there's just one little bug: When the menu gets too long on mobile (after having logged in), I can't scroll down. Because of that I can't reach the logout option.

**Comment by nmauersberg** [comment]
Thx for spotting this! It should be fixed now!

File: components/Section/index.tsx [hunk 1]
```diff
@@ -102,10 +103,21 @@ type SectionProps = {
 
 export const Section = ({ section }: SectionProps): ReactElement => {
   const { pageBuilderActive } = useContext(XbgeAppContext);
+  const { setShowModal } = useContext(OnboardingModalContext);
   const [modifiedSection, setModifiedSection] = useState<Section>(section);
   const [groupedElements, setGroupedElements] = useState<GroupedElements>([]);
   const router = useRouter();
 
+  // Some actions might be imported from the _actions.ts file,
```
Comment (nmauersberg) [inline components/Section/index.tsx:111]: Here is the change explained, regarding the dynamic action. @Valioesi

Comment (Valioesi) [reply components/Section/index.tsx:111]: Makes sense

**Review by schaerta — APPROVED** [review]
Thanks :)

File: layout/Header/MainMenu/MenuLink.tsx [hunk 2]
```diff
@@ -17,13 +18,17 @@ export const MenuLink = ({
...
+  extraCallback,
 }: MenuLinkProps) => {
   const prefixedSlug =
     entry.slug.substring(0, 1) === '/' ? entry.slug : `/${entry.slug}`;
 
   return (
     <Link key={entry.id} href={prefixedSlug}>
       <a
+        onClick={() => {
+          if (extraCallback) extraCallback();
```
Comment (Valioesi) [inline layout/Header/MainMenu/MenuLink.tsx:30]: Does this work in an a tag?

**Review by Valioesi — CHANGES_REQUESTED** [review]
The layout in the dropdowns seem kinda off (text not placed under the parent item). 
Other than that and the question I had, looks good!

File: layout/Header/style.module.scss [hunk 3]
```diff
@@ -1,5 +1,6 @@
 .header {
   max-height: $headerHeight;
+  min-height: $headerHeight;
```
Comment (Valioesi) [inline layout/Header/style.module.scss:3]: Why not just height? If min-height and max-height is the same, it's kinda redundant, or am I missing something?

File: layout/Header/MainMenu/MenuLink.tsx (hunk 2 above)
Comment (nmauersberg) [reply layout/Header/MainMenu/MenuLink.tsx:30]: Yes :) Fount this pattern on the internet, somewhere...

**Comment by nmauersberg** [comment]
> The layout in the dropdowns seem kinda off (text not placed under the parent item). Other than that and the question I had, looks good!

@Valioesi  What do you mean? For me it looks exactly as on the original page. I think the placement was on purpose.

**Comment by nmauersberg** [comment]
Ah, I thought you'd mean the content of the mobile menu.
Gotcha!

**Pull request merged**